torch_xla/experimental/xla_sharding.py

```diff
@@ -319,8 +319,28 @@ def _get_sharding_type(partition_spec: Tuple[Union[int, None]],
   return sharding_type
 
 
-def _get_tile_assignment(mesh: Mesh) -> List[int]:
-  return mesh.get_logical_mesh().tolist()
+def _get_tile_assignment(mesh: Mesh,
+                         partition_spec: Tuple[Union[int, None]]) -> List[int]:
+  # Use Torch.tensor here to make use of the torch.transpose_
+  mesh_list_tensor = torch.tensor(mesh.get_logical_mesh().tolist())
+  # This is 2d tensor 3d mesh case, tile_assigniment will be ignore in favor of
+  # group_assignment and replication_groups.
+  if (mesh_list_tensor.dim() != len(partition_spec)):
+    return mesh_list_tensor.tolist()
+  dim_to_current_dim = {}
+  for i in range(len(partition_spec)):
+    dim_to_current_dim[i] = i
+  for i in range(len(partition_spec)):
+    # we need transpose the mesh to match the partition_spec
+    if partition_spec[i] != None and partition_spec[i] != dim_to_current_dim[i]:
+      # need to find where the target dimension to transpose to
+      target_dim = dim_to_current_dim[partition_spec[i]]
+      mesh_list_tensor.transpose_(i, target_dim)
```

Comment:
I believe you can use permute instead of looping transpose.
https://pytorch.org/docs/stable/generated/torch.permute.html

Reply:
oh nice! let me try that.